core/store/src/trie/mod.rs

```diff
@@ -757,6 +760,97 @@ impl Trie {
     pub fn get_trie_nodes_count(&self) -> TrieNodesCount {
         self.storage.get_trie_nodes_count()
     }
+
+    /// Given a root and a key, it returns a MerkleProof for that key.
+    ///
+    ///  Whenever a proof is not found, an empty vector is returned.
+    ///
+    /// # Errors
+    ///
+    /// If the root is not found on the trie, a `StorageError` will be returned
```

Comment:
It is not just for looking up the root but any node in the path, no?

Reply:
What do you mean?

Reply:
```suggestion
    /// If the root or any subsequent is not found in the trie, `StorageError` will be returned
```